services/src/layers/storage.rs

```diff
@@ -18,47 +25,99 @@ pub enum LayerDbError {
     NoLayerForGivenId { id: LayerId },
 }
 
+pub const INTERNAL_PROVIDER_ID: LayerProviderId =
+    LayerProviderId::from_u128(0xce5e_84db_cbf9_48a2_9a32_d4b7_cc56_ea74);
+
+pub const INTERNAL_LAYER_DB_ROOT_COLLECTION_ID: Uuid =
+    Uuid::from_u128(0x0510_2bb3_a855_4a37_8a8a_3002_6a91_fef1);
+
 #[async_trait]
-pub trait LayerDb: Send + Sync {
-    async fn add_layer(&self, layer: Validated<AddLayer>) -> Result<LayerId>;
-    async fn add_layer_with_id(&self, id: LayerId, layer: Validated<AddLayer>) -> Result<()>;
+/// Storage for layers and layer collections
+pub trait LayerDb: LayerCollectionProvider + Send + Sync {
+    /// add new `layer` to the given `collection`
+    async fn add_layer(
+        &self,
+        layer: Validated<AddLayer>,
+        collection: &LayerCollectionId,
+    ) -> Result<LayerId>;
 
-    async fn get_layer(&self, id: LayerId) -> Result<Layer>;
+    /// add new `layer` with fixed `id` to the given `collection`
+    /// TODO: remove this method and allow stable names instead
```

Comment:
what is this todo about?

Reply:
right now we often have two methods to add something (a dataset, a layer, a provider) to the Db. One, where the Id of the newly created element is automatically assigned by the Db (as it should be) and one, where we pass the Id from the outside (because we need it to always be the same). This comment is about removing the latter one once when have stable names for items, that make stable ids obsolete.